### ISSUE TITLE
forrtl: severe (174): SIGSEGV, segmentation fault occurred using molecular dynamics and GFN-FF 

### ISSUE THREAD

**Issue by msgroi2** [grimme-lab/xtb#1007]
Dear developers,
my name is Mauro Sgroi and I am trying to use xtb to simulate solid state sulfide ionic conductors.
I get the error in the object running xtb on the computational server at my university.
I already set ulimit -s unlimited and tried different values of OMP_STACKSIZE (from 1G to 128G).
OMP_NUM_THREADS was set to 1 (the same error can be reproduced with other values for the number of threads).
My launch command is xtb --verbose --mcgfnff --md 2x2x2.poscar
The error is reported below:
 Periodic properties

           -------------------------------------------------
          | TOTAL ENERGY              -31.545683963545 Eh   |
          | GRADIENT NORM               0.109949584694 Eh/α |
           -------------------------------------------------
           -------------------------------------------------
          |               Molecular Dynamics                |
           -------------------------------------------------
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
xtb                000000000046E52A  Unknown               Unknown  Unknown
xtb                0000000002CF6140  Unknown               Unknown  Unknown
xtb                0000000000680707  xtb_gfnff_shake_m          49  shake_module.f90
xtb                000000000042E7BB  xtb_prog_main_mp_        1100  main.F90
xtb                0000000000450B42  MAIN__                     57  primary.f90
xtb                0000000000401E42  Unknown               Unknown  Unknown
xtb                0000000002CECD8A  Unknown               Unknown  Unknown
xtb                0000000002CEE627  Unknown               Unknown  Unknown
xtb                0000000000401D05  Unknown               Unknown  Unknown

The same attached poscar file works for geometry optimization.
How can I solve this issue?
Thanks a lot and best regards,
Maurio Sgroi.

**Comment by Albkat** [comment]
hello, can you share your input geometry?

**Comment by msgroi2** [comment]
ah, sorry. I forgot it.

[2x2x2.txt](https://github.com/grimme-lab/xtb/files/14888066/2x2x2.txt)

**Comment by msgroi2** [comment]
Dear all,
any idea about my problem?
Thanks a lot in advance!
Best regards,
Mauro.

**Comment by Thomas3R** [comment]
Dear Mauro,

You need to include your MD settings in a xcontrol file and parse it to xtb via the command line input. E.g.:
xtb 2x2x2.poscar --mcgfnff --md --input xcontrol
Though it is not an issue (in poscar format), I would not recommend including the MD settings in your .poscar file.

Is there a reason why the unit cell is so much larger than the contained cluster of atoms? This seems unreasonable and the MD with the force field destroys the cluster as one would expect. Below, I appended my results on your system for an MD with standard GFN-FF settings set in the xcontrol file. I manually terminated the MD after some time.
[mcgfnff_md_xtb670.zip](https://github.com/grimme-lab/xtb/files/14947412/mcgfnff_md_xtb670.zip)

To get these results, I downloaded xtb-6.7.0-linux-x86_64.tar.xz from https://github.com/grimme-lab/xtb/releases/tag/v6.7.0
Then ran the MD with
/path/to/the/xtb670-binary 2x2x2.poscar --mcgfnff --md --input xcontrol > 0ffmd.out

As a last note: The GFN Force Field is not tailored for ionic crystals and the mcGFN-FF is specifically designed for molecular crystals. Of course, it would be interesting to see how the force field performs on such a system. 

I hope this helps you to run your own MDs with our force field.

Best,
Thomas

**Comment by msgroi2** [comment]
Dear Thomas,
thanks a lot.
I am implementing your suggestions. The poscar file I attached had an error (now I substituted the cell parameter with about 20 Angstroms).  Sorry!

On the other hand now, I get the following error:

OMP: Error #34: System unable to allocate necessary resources for OMP thread:
OMP: System error #11: Resource temporarily unavailable
OMP: Hint Try decreasing the value of OMP_NUM_THREADS.
forrtl: error (76): Abort trap signal

I am launching with OMP_NUM_THREADS=4 on a machine with 24 cores.
Best regards,
Mauro.
